internal/core/src/rescores/Scorer.h

```diff
@@ -41,11 +74,109 @@ class WeightScorer : public Scorer {
         return filter_;
     }
 
+    void
+    set_score(std::optional<float>& old_score,
```

Comment:
make private, and add comment

also make `random_score()` function of `RandomScorer` private.